### PR TITLE
fix: fix tasks filter issue

### LINKED ISSUE #844: Remember the filter for the list pages
the experience today is that you go to the tasks page (or dashboards page), use the filter to find the tasks you want, then click on one of them. When you click the back button in the browser, you are back to the page with no filters in place.

Users can have 100s of tasks, so i'd love to be able to save that filter and get back to it.

I propose we find a way to keep the filter in place when a user clicks the back button specifically for the tasks page.

### PR DESCRIPTION
**Pull request by subirjolly**
Closes #844

### CODE REVIEW

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 60 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1147)

File: cypress/e2e/cloud/tasklist.test.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import {Organization} from '../../../src/types'
+
+describe('Checkout Page', () => {
```
Comment (Asalem1) [inline cypress/e2e/cloud/tasklist.test.ts:3]: checkout page? We should be able to put this all in the `tasks.test.ts` file

File: cypress/e2e/cloud/tasklist.test.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  beforeEach(() => {
+    cy.flush()
+
+    cy.signin().then(() => {
+      cy.get('@org').then(({id}: Organization) => {
+        console.log('Organization:::: ', id)
```
Comment (Asalem1) [inline cypress/e2e/cloud/tasklist.test.ts:9]: no need for this

File: cypress/e2e/cloud/tasklist.test.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        .clear()
+        .type(task.offset)
+      cy.getByTestID('flux-editor').type(task.query)
+
+      // Save Task
+      cy.getByTestID('task-save-btn').click()
+    })
+
+    // Search for a task
+    cy.getByTestID('search-widget').type(tasks[0].name)
```
Comment (Asalem1) [inline cypress/e2e/cloud/tasklist.test.ts:54]: should we include partial matches?

File: cypress/e2e/cloud/tasklist.test.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      .children()
+      .should('have.length', 1)
+    cy.getByTestID('resource-list--body')
+      .children()
+      .getByTestID('task-card--name')
+      .click()
+
+    // Navigate away from current page back to Tasks List page
+    cy.getByTestID('task-cancel-btn').click()
+    cy.getByTestID('search-widget').should('have.value', tasks[0].name)
```
Comment (Asalem1) [inline cypress/e2e/cloud/tasklist.test.ts:65]: Now that we've validated that the search is populated when navigating back to the task list, we can also validate that the tasks rendered are filtered

File: cypress/e2e/cloud/tasklist.test.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      .children()
+      .should('have.length', 1)
+    cy.getByTestID('resource-list--body')
+      .children()
+      .getByTestID('task-card--name')
+      .click()
+
+    // Navigate away from current page back to Tasks List page
+    cy.getByTestID('task-cancel-btn').click()
+    cy.getByTestID('search-widget').should('have.value', tasks[1].name)
```
Comment (Asalem1) [inline cypress/e2e/cloud/tasklist.test.ts:81]: same as above

**Review by Asalem1 — CHANGES_REQUESTED** [review]
I'd love to see a gif of this to validate that this works.

File: cypress/e2e/cloud/tasklist.test.ts (hunk 1 above)
Comment (subirjolly) [reply cypress/e2e/cloud/tasklist.test.ts:3]: Copy Pasta

File: cypress/e2e/cloud/tasklist.test.ts (hunk 2 above)
Comment (subirjolly) [reply cypress/e2e/cloud/tasklist.test.ts:9]: debugging. 👍

File: cypress/e2e/cloud/tasklist.test.ts (hunk 3 above)
Comment (subirjolly) [reply cypress/e2e/cloud/tasklist.test.ts:54]: sure. I will add another case.

File: cypress/e2e/cloud/tasklist.test.ts (hunk 4 above)
Comment (subirjolly) [reply cypress/e2e/cloud/tasklist.test.ts:65]: makes sense

File: cypress/e2e/cloud/tasklist.test.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
+import {Organization} from '../../../src/types'
+
+describe('Tasks List Page', () => {
```
Comment (Asalem1) [inline cypress/e2e/cloud/tasklist.test.ts:3]: this entire test should live in `tasks.test.ts`

**Comment by subirjolly** [comment]
Kapture 2021-04-08 at 15 07 34

**Review by Asalem1 — APPROVED** [review]
Dope. Way easier than I thought it would be. I think @russorat mentioned something about navigating to different parts of the app, then navigating back. I'm not quite sure what the specs were but if it's good with him then we should be all set

**Pull request merged**